scripts/cangen/cangen/traffic.py

```diff
@@ -0,0 +1,48 @@
+"""
+Author: Manush Patel
+Date: 2024-11-24
+https://macformula.github.io/racecar/firmware/can-traffic/
+"""
+
+from dataclasses import dataclass
+
+
+@dataclass
+class CANMessage:
+    data_length: int
+    frequency: int
+
+    @property
+    def bit_count(self):
+        fixed_length = 44  # fixed length of CAN frame
+        variable_length = self.data_length * 8.0  # accounted for bit stuffing
+        total_frame_length = variable_length + fixed_length
+        return total_frame_length
+
+
+def calculate_bus_load(messages: list[CANMessage], can_speed: int) -> float:
+    """Calculates the bus load percentage based on CAN messages and bus speed
+
+    Parameters
+    ----------
+    messages : list[CANMessage]
+        A list of CANMessage objects, each containing `bit_count` and `frequency` attributes
+    can_speed : int
+        The CAN bus speed in bits per second (bps).
+
+    Returns
+    -------
+    float
+        Returns bus load as a percentage
+    """
+    if can_speed < 0:
```

Comment:
This is bad practice. We will likely use your function to terminate the build process if the traffic is too high, but this line will incorrectly say that there is 0 bus load if the speed is `<0`

we should raise an error if less than __or equal__ to 0, not just if less than

Instead, you should raise an error. This will terminate the build process because an invalid value was provided.

The correct error type is `ValueError`. You should replace this block with

```python
if can_speed <= 0:
    raise ValueError(f"Invalid CAN speed {can_speed}. Must be positive.")
```